warehouse/validations/validate.go

```diff
@@ -17,378 +20,537 @@ import (
-		return
+	filePath = fmt.Sprintf("%v/%v/%v.%v.%v.%v",
+		tmpDirPath,
+		connectionTestingFolder,
+		destinationType,
+		warehouseutils.RandHex(),
+		time.Now().Unix(),
+		warehouseutils.GetLoadFileFormat(destinationType),
+	)
+	if err = os.MkdirAll(filepath.Dir(filePath), os.ModePerm); err != nil {
+		return fmt.Errorf("create directory: %w", err)
 	}
 
-	// creating temporary file
-	testFile, err := os.Create(testFilePath)
-	if err != nil {
-		pkgLogger.Errorf("DCT: Failed to create tempFilePath: %s with error: %s", testFilePath, err.Error())
-		return
+	if downloadFile, err = os.Create(filePath); err != nil {
+		return fmt.Errorf("creating file: %w", err)
 	}
 
 	// cleanup
-	defer misc.RemoveFilePaths(testFilePath)
-	defer func() { _ = testFile.Close() }()
+	defer misc.RemoveFilePaths(filePath)
+	defer func() { _ = downloadFile.Close() }()
 
-	// downloading temporary file to specified from object storage location
-	err = fm.Download(context.TODO(), testFile, location)
+	if err = fm.Download(context.TODO(), downloadFile, location); err != nil {
+		return fmt.Errorf("downloading file: %w", err)
+	}
+	return nil
+}
+
+func createFileManager(dest *backendconfig.DestinationT) (filemanager.FileManager, error) {
+	var (
+		destType = dest.DestinationDefinition.Name
+		conf     = dest.Config
+		provider = warehouseutils.ObjectStorageType(destType, conf, misc.IsConfiguredToUseRudderObjectStorage(conf))
+	)
+
+	fileManager, err := fileManagerFactory.New(&filemanager.SettingsT{
+		Provider: provider,
+		Config: misc.GetObjectStorageConfig(misc.ObjectStorageOptsT{
+			Provider:         provider,
+			Config:           conf,
+			UseRudderStorage: misc.IsConfiguredToUseRudderObjectStorage(conf),
+			WorkspaceID:      dest.WorkspaceID,
+		}),
+	})
 	if err != nil {
-		pkgLogger.Errorf("DCT: Failed to download tempFilePath: %s with error: %s", location, err.Error())
-		return
+		return nil, fmt.Errorf("creating file manager: %w", err)
 	}
-	return
+
+	fileManager.SetTimeout(objectStorageValidationTimeout)
+
+	return fileManager, nil
 }
 
-func (ct *CTHandleT) loadTable(loadFileLocation string) (err error) {
-	destination := ct.infoRequest.Destination
-	destinationType := destination.DestinationDefinition.Name
+func createManager(dest *backendconfig.DestinationT) (manager.WarehouseOperations, error) {
+	var (
+		destType  = dest.DestinationDefinition.Name
+		warehouse = createDummyWarehouse(dest)
+
+		operations manager.WarehouseOperations
+		err        error
+	)
 
-	stagingTableName := stagingTableName()
+	if operations, err = manager.NewWarehouseOperations(destType); err != nil {
+		return nil, fmt.Errorf("getting manager: %w", err)
+	}
 
-	// Create table
-	err = ct.manager.CreateTable(stagingTableName, TestTableSchemaMap)
-	if err != nil {
-		return
+	operations.SetConnectionTimeout(warehouseutils.TestConnectionTimeout)
+
+	if err = operations.Setup(warehouse, &mockUploader{
+		dest: dest,
+	}); err != nil {
```

Comment:
I am not sure why we are doing things. Since `mockUploader` is usually for helping testing doesn't imply a lot about its functionality.

Reply:
True. Should I rename this to something like `dummyUploader`?

Reply:
We wanted 2 functionalities from Uploader interface `GetLoadFileType` and `UseRudderStorage`

Reply:
I am thinking of using noop uploader to avoid adding default implementations.
https://github.com/rudderlabs/rudder-server/pull/2993

Reply:
Once the above PR merges, we would not need this.

Reply:
Because of the complexity of the changes associated with `Uploader` interface. I am thinking to move to a separate PR.
I renamed it to `dummyUploader` for the current use case.